addons/website/static/src/snippets/s_rating/options.js

```diff
@@ -0,0 +1,184 @@
+odoo.define('website.s_rating_options', function (require) {
+'use strict';
+
+const ColorpickerDialog = require('web.ColorpickerDialog');
+var weWidgets = require('wysiwyg.widgets');
+var options = require('web_editor.snippets.options');
+
+options.registry.Rating = options.Class.extend({
+    /**
+     * @override
+     */
+    start: function () {
+        this.faClassActiveCustomIcons = '';
+        this.faClassInactiveCustomIcons = '';
+        this.iconType = this.$target.attr('data-icon');
+        this.activeIconColor = this._ensureCssColor(this.$target.attr('data-active-icon-color'));
+        this.inactiveIconColor = this._ensureCssColor(this.$target.attr('data-inactive-icon-color'));
+        if (this.iconType === 'custom') {
+            this.faClassActiveCustomIcons = this.$target.attr('data-active-custom-icon');
+            this.faClassInactiveCustomIcons = this.$target.attr('data-inactive-custom-icon');
+        }
+        return this._super.apply(this, arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Display the selected icon type.
+     *
+     * @see this.selectClass for parameters
+     */
+    setIcons: function (previewMode, widgetValue, params) {
+        this.iconType = widgetValue;
+        this._renderIcons();
+        this.$target.attr('data-icon', widgetValue);
+        this.$target.attr('data-active-custom-icon', "");
+        this.$target.attr('data-inactive-custom-icon', "");
+    },
+
+    /**
+     * Allows to select a font awesome icon with media dialog.
+     *
+     * @see this.selectClass for parameters
+     */
+    customIcon: async function (previewMode, widgetValue, params) {
+        return new Promise(resolve => {
+            const dialog = new weWidgets.MediaDialog(
+                this,
+                {noImages: true, noDocuments: true, noVideos: true, mediaWidth: 1920},
+                $('<i/>')
+            );
+            this._saving = false;
+            dialog.on('save', this, function (attachments) {
+                this._saving = true;
+                const $icons = this.$target.find('.rating_icon');
+                _.each($icons, el => {
+                    if (el.dataset.iconActive === params.customActiveIcon) {
+                        $(el).removeClass().addClass('rating_icon fa-fw ' + attachments.className);
```

Comment:
Don't remove all classes, only the fa ones (what if another option adds a class on fa icons ?)